test/suites/metrics.sh

```diff
@@ -39,12 +42,18 @@ test_metrics() {
   # c1 metrics should show as the container is running
   curl -k -s --cert "${TEST_DIR}/metrics.crt" --key "${TEST_DIR}/metrics.key" -X GET "https://${metrics_addr}/1.0/metrics" | grep "name=\"c1\""
 
-  # c2 metrics should show the container as stopped
-  curl -k -s --cert "${TEST_DIR}/metrics.crt" --key "${TEST_DIR}/metrics.key" -X GET "https://${metrics_addr}/1.0/metrics" | grep "name=\"c2\"" | grep "state=\"STOPPED\""
+  # c2 metrics should not exist as it's not running
+  ! curl -k -s --cert "${TEST_DIR}/metrics.crt" --key "${TEST_DIR}/metrics.key" -X GET "https://${metrics_addr}/1.0/metrics" | grep "name=\"c2\"" || false
 
   # make sure no other endpoint is available
   curl -k -s --cert "${TEST_DIR}/metrics.crt" --key "${TEST_DIR}/metrics.key" -X GET "https://${metrics_addr}/1.0/instances" | grep "\"error_code\":404"
 
+  # Check that we can get the count of existing containers. There should be two: c1 (RUNNING) and c2 (STOPPED).
+  curl -k -s --cert "${TEST_DIR}/metrics.crt" --key "${TEST_DIR}/metrics.key" -X GET "https://${metrics_addr}/1.0/metrics" | grep -E '^(lxd_containers)\{.*\} 2' || false
```

Comment:
This looks like a copy-n-paste of the older `grep -E ... || false` variant. Please use the same form as in the previous check further up in that same file.

Reply:
fixed